lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -108,11 +108,149 @@ class RsqrtOpConversion : public OpConversionPattern<tosa::RsqrtOp> {
+
+      auto broadcastArg = rewriter.create<emitc::CallOp>(
+          srcOp->getLoc(), srcOp.getType(), broadcastCallee, broadcastArgs,
+          templateBroadcastArgs, operand);
+      // Replace the original operand with the result of the broadcast_in_dim
+      // operation
+      broadcastedOperands[i] = broadcastArg.getResult(0);
+    }
+  }
+  return broadcastedOperands;
+}
+
+template <typename SrcOp>
+class CallOpBroadcastableConversion : public OpConversionPattern<SrcOp> {
+  using OpConversionPattern<SrcOp>::OpConversionPattern;
+
+public:
+  CallOpBroadcastableConversion(MLIRContext *ctx, StringRef funcName,
+                                bool explicitResultType = false,
+                                bool explicitOperandTypes = false)
+      : OpConversionPattern<SrcOp>(ctx), funcName(funcName),
+        explicitResultType(explicitResultType),
+        explicitOperandTypes(explicitOperandTypes) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(SrcOp srcOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    StringAttr callee = rewriter.getStringAttr(funcName);
+    ArrayAttr args;
+
+    SmallVector<Attribute, 4> templateArgs_;
+
+    if (explicitResultType) {
+      Type type = srcOp.getType();
+      templateArgs_.push_back(TypeAttr::get(type));
+    }
+
+    if (explicitOperandTypes) {
+      for (auto &operand : operands) {
+        Type type = operand.getType();
+        templateArgs_.push_back(TypeAttr::get(type));
+      }
+    }
+
+    ArrayAttr templateArgs = ArrayAttr::get(srcOp.getContext(), templateArgs_);
+
+    SmallVector<Value, 2> broadcastedOperands =
+        create_broadcast_op_if_needed(srcOp, operands, rewriter);
+
+    rewriter.replaceOpWithNewOp<emitc::CallOp>(
+        srcOp, srcOp.getType(), callee, args, templateArgs,
+        mlir::ValueRange({broadcastedOperands[0], broadcastedOperands[1]}));
+
+    return success();
+  }
+
+  StringRef funcName;
+  // If set, use the result type of the operation as template parameter
+  bool explicitResultType;
+  // If set, use the operand types as (additional) template parameters
+  bool explicitOperandTypes;
+};
+
+class MulOpConversion : public OpConversionPattern<tosa::MulOp> {
+  using OpConversionPattern<tosa::MulOp>::OpConversionPattern;
+
+public:
+  MulOpConversion(MLIRContext *ctx, StringRef funcName,
+                  bool explicitResultType = false,
+                  bool explicitOperandTypes = false)
+      : OpConversionPattern<tosa::MulOp>(ctx) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(tosa::MulOp mulOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    StringRef funcName = "tosa::mul";
+    StringAttr callee = rewriter.getStringAttr(funcName);
+
+    SmallVector<Attribute, 1> args_;
+    args_.push_back(rewriter.getIndexAttr(0));
+    args_.push_back(rewriter.getIndexAttr(1));
+    args_.push_back(mulOp.shiftAttr());
```

Comment:
Just add 'shiftAttr' if needed. See my comment above: we can force the compiler to choose the right function.